cmake/arm-cross.cmake

```diff
@@ -6,3 +6,71 @@ set(CMAKE_SYSTEM_PROCESSOR aarch64)
 set(CMAKE_C_COMPILER aarch64-linux-gnu-gcc)
 set(CMAKE_CXX_COMPILER aarch64-linux-gnu-g++)
 set(CMAKE_C_COMPILER_ID GNU)
+
+# Allow the developer to target a runtime environment other than that of the
+# build host.
+if (OE_PROGRAM_PATH AND OE_SYSROOT)
+    set(CMAKE_PROGRAM_PATH ${OE_PROGRAM_PATH})
+    set(CMAKE_SYSROOT ${OE_SYSROOT})
+    set(CMAKE_FIND_ROOT_PATH ${OE_SYSROOT})
+    set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
+    set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)
+    set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
+    set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+    set(ENV{PKG_CONFIG_SYSROOT_DIR} ${OE_SYSROOT})
+endif ()
+
+# When using GCC to compile assembly files.
+set(OE_TZ_TA_S_FLAGS
+    -DASM=1
+    -pipe)
+
+# When using GCC to compile C/CXX files.
+set(OE_TZ_TA_C_FLAGS
+    -DARM64=1
+    -D__LP64__=1
+    -mstrict-align
+    -nostdinc
+    -nostdlib
+    -nodefaultlibs
+    -nostartfiles
+    -fno-builtin-memcpy
+    -fno-builtin-memset
+    -ffreestanding
+    -funwind-tables
+    -fpie
+    -fPIC
+    -gdwarf)
+
+string(REPLACE ";"
```

Comment:
Maybe leave a comment as to what we're doing here for non-CMake people (converting a CMake list variable to a string variable for direct use by a compilation tool, also, I think I have an example of how to do this without `REPLACE` somewhere that I'll try to dig up...)

Reply:
It turns out that this variable isn't used anywhere, so I've removed it.